File: assets/nft.js
// var fs = require('fs');
// for (var i =0 ; i < 250; i++) {
//   var json = {}
//   json.name = "SCGT" ;
//   json.description = "SCGT Constellation";
//   json.image = "https://raw.githubusercontent.com/gpteth/constellation/refs/heads/main/assets/" + i + ".png";
//   json.external_url = "https://solana.com/zh";
//   json.attributes =  [  { "trait_type": "constellation", "value": "Taurus" }, { "trait_type": "trait2", "value": "value2" } ];
//   json.properties = {
//     files: [
//       {
//         uri: "https://raw.githubusercontent.com/gpteth/constellation/refs/heads/main/assets/" + i + ".png",
//         type: "image/png"
//       }
//     ],
//     category: "image"
//   };
//   fs.writeFileSync(i + '.json', JSON.stringify(json, null, 2));
// }

// {
//   "name": "SCGT",
//   "description": "SCGT Constellation",
//   "image": "https://raw.githubusercontent.com/gpteth/constellation/refs/heads/main/assets/1.png",
//   "external_url": "https://solana.com/zh",
//   "attributes": [
//       {
//           "trait_type": "constellation",
//           "value": "Capricorn"
//       },
//       {
//           "trait_type": "trait2",
//           "value": "value2"
//       }
//   ],
//   "properties": {
//       "files": [
//           {
//               "uri": "https://raw.githubusercontent.com/gpteth/constellation/refs/heads/main/assets/1.png",
//               "type": "image/png"
//           }
//       ],
//       "category": "image"
//   }
// }
// var fs = require('fs');

// for (var i = 0; i < 19; i++) {
//   var json = {}
//   json.name = "Token #" + i;
//   json.description = "This is the description for token #" + i;
//   json.image = "ipfs://QmUzeXJiSu8tBAejX5EHBG9jFHDibpXwETdDDiSLE6U6SB/" + i + ".png";

//   fs.writeFileSync( i + '.json', JSON.stringify(json));
// }


// const fs = require('fs');
// const path = require('path');

// 要重命名的文件夹
// const dir = 'E:\\文件\\Tiara/';
// const dir = './';

// // 读取文件夹下所有文件
// fs.readdir(dir, (err, files) => {
//     if (err) {
//         console.log(err);
//         return;
//     }

//     // 遍历文件夹下所有文件，并重命名
//     for (let i = 1; i < files.length -1; i++) {
//         const oldPath = path.join(dir, files[i]);
//         const newPath = path.join(dir, `${i}.json`);

//         // 修改文件名
//         fs.rename(oldPath, newPath, err => {
//             if (err) {
//                 console.log(err);
//                 return;
//             }
//         });
//     }
// });


// // {
// //     "name": "Random Name" + Math.floor(Math.random() * 100),
// //     "age": Math.floor(Math.random() * 100),
// //     "address": "Random Address" + Math.floor(Math.random() * 100),
// //     "phone": Math.floor(Math.random() * 100) + "-" + Math.floor(Math.random() * 10000)
// // }

// // {
// //     "description": "Friendly OpenSea Creature that enjoys long swims in the ocean.", 
// //     "external_url": "https://openseacreatures.io/3", 
// //     "image": "https://storage.googleapis.com/opensea-prod.appspot.com/puffs/3.png", 
// //     "name": "Dave Starbelly",
// //     "attributes": [ ... ]
// //   }

File: assetsjson/nft.py
import os
import json

# 设置 JSON 文件夹路径
json_folder = 'nft_json'  # 存放 JSON 文件的文件夹

# 确保 JSON 文件夹存在
if not os.path.exists(json_folder):
    os.makedirs(json_folder)

# 星座列表，按顺序循环
constellations = [
    "Capricorn", "Aquarius", "Aries", "Leo", "Libra", "Scorpio",
    "Cancer", "Sagittarius", "Pisces", "Virgo", "Gemini", "Taurus"
]

# 开始生成 JSON 文件
for idx in range(3000):
    constellation = constellations[idx % len(constellations)]  # 循环取星座
    
    nft_data = {
        "name": "SCGT",
        "description": "SCGT Constellation",
        "image": f"https://raw.githubusercontent.com/gpteth/constellation2/refs/heads/main/assets/{(idx % 12) + 1}.png",
        "external_url": "https://solana.com/zh",
        "attributes": [
            {
                "trait_type": "constellation",
                "value": constellation
            },
            {
                "trait_type": "trait2",
                "value": "value2"
            }
        ],
        "properties": {
            "files": [
                {
                    "uri": f"https://raw.githubusercontent.com/gpteth/constellation2/refs/heads/main/assets/{(idx % 12) + 1}.png",
                    "type": "image/png"
                }
            ],
            "category": "image"
        }
    }
    
    # 保存 JSON 文件
    json_filename = f"{idx+1}.json"
    json_path = os.path.join(json_folder, json_filename)
    
    with open(json_path, 'w') as json_file:
        json.dump(nft_data, json_file, indent=4)

print(f"✅ 已成功生成 3000 个 NFT JSON 文件！")